proxy/http/HttpTransact.cc

```diff
@@ -669,6 +669,11 @@ HttpTransact::StartRemapRequest(State *s)
 
     s->hdr_info.client_request.set_url_target_from_host_field();
 
+    // Since we're not doing remap, we still have to allow for these overridable
+    // configurations to modify follow-redirect behavior. Someone could for example
+    // have set them in a plugin other than conf_remap running in a prior hook.
+    s->state_machine->enable_redirection = (s->txn_conf->number_of_redirections > 0);
```

Comment:
So I poked at this, and technically it's not needed, but removing it makes the code a lot more fragile. The issue is that enable_redirection is used not only as a flag to track the number_of_redirects setting, it also effectively turns off this feature when necessary. Removing it really makes the code a lot less readable (and more fragile I think).